hibernate-core/src/main/java/org/hibernate/sql/results/internal/domain/CircularBiDirectionalFetchImpl.java

```diff
@@ -182,11 +181,14 @@ public Object assemble(RowProcessingState rowProcessingState, JdbcValuesSourcePr
 			final Object initializedInstance = initializer.getInitializedInstance();
 			final LazyInitializer lazyInitializer = HibernateProxy.extractLazyInitializer( initializedInstance );
 			if ( lazyInitializer != null ) {
-				if ( initializedInstance.getClass().isAssignableFrom( javaType.getJavaTypeClass() ) ) {
+				initializer.initializeInstance( rowProcessingState );
```

Comment:
I think we should only do that if `fetchTiming != EAGER`

Reply:
fixed